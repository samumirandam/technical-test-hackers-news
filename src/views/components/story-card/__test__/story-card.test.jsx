import React from "react";
import { render, screen, fireEvent } from "@utils/test-utils";

import StoryCard from "../index";

const defaultProps = {
  created_at: "2002-08-26T00:27:26.000Z",
  author: "didericis",
  story_title: "Threats of Blackouts Drive Japan to Embrace Nuclear Again",
  story_url:
    "https://financialpost.com/pmn/business-pmn/threats-of-blackouts-drive-japan-to-embrace-nuclear-again",
};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<StoryCard {...setupStore} />, {});
};

describe("Test for StoryCard component", () => {
  test("Should render without errors", () => {
    setup();
    expect(
      screen.getByText(
        "Threats of Blackouts Drive Japan to Embrace Nuclear Again"
      )
    ).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test("Should render click in story", () => {
    setup();
    fireEvent.click(
      screen.getByText(
        "Threats of Blackouts Drive Japan to Embrace Nuclear Again"
      )
    );
  });
});
