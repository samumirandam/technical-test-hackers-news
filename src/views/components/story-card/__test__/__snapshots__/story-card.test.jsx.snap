// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for StoryCard component Should renders the same component 1`] = `
<div>
  <div
    class="StoryCard"
  >
    <button
      class="Button StoryCard__content"
      type="button"
    >
      <div
        class="StoryCard__data"
      >
        <figure
          class="StoryCard__time-image"
        >
          <svgrurl />
        </figure>
        <p>
          about 20 years by didericis
        </p>
      </div>
      <p
        class="StoryCard__title"
      >
        Threats of Blackouts Drive Japan to Embrace Nuclear Again
      </p>
    </button>
    <button
      class="Button StoryCard__favorite noStyles"
      type="button"
    >
      <figure
        data-testid="favorite-button"
      >
        <svgrurl />
      </figure>
    </button>
  </div>
</div>
`;
